ci/scripts/install_pandas.sh

```diff
@@ -36,7 +36,7 @@ else
 fi
 
 if [ "${pandas}" = "upstream_devel" ]; then
-  pip install git+https://github.com/pandas-dev/pandas.git --no-build-isolation
+  pip install git+https://github.com/pandas-dev/pandas.git
```

Comment:
I wonder why we did use `--no-build-isolation` in the first place. LGTM

Reply:
I think just to be more efficient (all dependencies like numpy and cython were already installed in the env (to build pyarrow), so no need to install those in an isolation build environment). But now pandas started to depend on `versioneer` as build-time dependency (instead of vendoring it), so this is the cleanest solution (otherwise we would have to install versioneer first)